File: Java/SingleNumber.java
class Solution {
    public int singleNumber(int[] nums) {
        
        
        HashMap<Integer,Integer> map=new HashMap<>();
        int ans=0;
        for(int i=0;i<nums.length;i++)
        {
            map.put(nums[i],0);
        }
        int count=0;
        for(int i=0;i<nums.length;i++)
        {
            if(map.containsKey(nums[i]))
            {
                map.put(nums[i],map.get(nums[i])+1);
            } 
        }
        for(int i=0;i<nums.length;i++)
        {
            if(map.get(nums[i])==1)
                ans=nums[i];
        }
        return ans;
        
        /*int xor=nums[0]; //Faster approach
        for(int i=1;i<nums.length;i++)
        {
            xor=xor^nums[i];
        }
        return xor;*/
        
       
    }
}

File: Python/SingleNumber.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        my_dict=defaultdict(int)
        
        '''for num in nums:
            my_dict[num]=my_dict.get(num,0)+1'''
        
        for num in nums:
            my_dict[num]+=1
        
        
        for i in my_dict:
            if my_dict[i] == 1:
                return i
            
    '''
    for i in nums:
        if nums.count(i)==1:
            return i
    '''